src/main/java/core/basesyntax/model/Report.java

```
@@ -0,0 +1,46 @@
+package core.basesyntax.model;
+
+import java.math.BigInteger;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.BiConsumer;
+import java.util.function.Supplier;
+
+public final class Report {
+    private final Map<FruitType, BigInteger> recordMap;
+
+    private Report(Map<FruitType, BigInteger> recordMap) {
+        this.recordMap = Collections.unmodifiableMap(Objects.requireNonNull(recordMap));
+    }
+
+    public void consume(BiConsumer<FruitType, BigInteger> consumer) {
+        recordMap.forEach(consumer);
```

Missing null-validation for consumer parameter. `public void consume(BiConsumer<FruitType, BigInteger> consumer)` (line 18) invokes `recordMap.forEach(consumer)` without checking `consumer` for null. This violates the generalized null-handling requirement: "Null handling: validate method arguments ... and throw informative RuntimeException if violated." Add `Objects.requireNonNull(consumer, "consumer must not be null")` (or an equivalent RuntimeException with informative message) at the start of `consume` to fail fast and give clear diagnostics.